### PR TITLE
Blocking Users

### PR DESCRIPTION
**Pull request by maxwell1231**
## Overview

Created routes to block users, find all blocked users, and unblock users. The routes are GET api/block, POST api/block, and POST api/unblock.

## Changes Made

Created a blocked_users assoc table in the Person model. When blocking and unblocking with POST requests, you need to specify the user_id using "blocked_user" and "unblocked_user" as the key, respectively. 

## Test Coverage

Tested with postman requests.

### CODE REVIEW

File: src/person/controllers/block_controller.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        self._serializer = serializer
+
+    def get_blocked_users(self, user):
+        """Get blocked users for `user`."""
+        return user.person.blocked_users.all()
+
+    def unblock(self, user, unblocked_user):
+        """Unblock `blocked_user` for `user`."""
+        person = user.person
+        # if unblocked_user in person.blocked_users.all():
```
Comment (joycrwu) [inline src/person/controllers/block_controller.py:69]: let's get rid of this commented code

**Review by joycrwu — COMMENTED** [review]
looks good, just remove the lines of comments!

**Review by joycrwu — COMMENTED** [review]
does this implementation account for storing the information of all of the users that have blocked this particular user?

**Comment by maxwell1231** [comment]
No, should we do that?

**Review by joycrwu — APPROVED** [review]

**Review by patriciaahuang — APPROVED** [review]

**Pull request merged**